### DISCUSSION TITLE
Edit HMI Page Files for an inverted Display

### DISCUSSION THREAD

**Discussion by mimk97** [HASwitchPlate/HASPone#251] [Q&A]
Hi.
is it possible to edit the Page Files with Nextion Editor and rotate it 180 Degrees?
I use inverted Sscreen

**Comment by aderusha** [comment]
Take a look at the -Inverted firmware, eg: https://github.com/HASwitchPlate/HASPone/blob/main/Nextion_HMI/HASwitchPlate-Inverted.tft

**Reply by mimk97** [reply]
hhm. can i extract the TFT, so i can edit the HMI?

**Reply by aderusha** [reply]
If you're editing the HMI in Nextion editor, you can rotate the project file in the device setup.  
![CXOvDPc](https://github.com/user-attachments/assets/37e5ddd5-6ab1-4058-9405-c125beb6780d)

**Reply by mimk97** [reply]
yes. i done it. But then the Touch doesn´t work :(
isn´t it neccessary to new Program the Object´s ?

**Reply by mimk97** [reply]
i edited Page 3, see attached. then Rotate Display 180 Degrees and the export as TFT
![Screenshot 2024-11-11 153003](https://github.com/user-attachments/assets/a6dc54cf-843c-4415-98d7-4690764fbb92)
![Screenshot 2024-11-11 153031](https://github.com/user-attachments/assets/e10501e8-9a0c-45a2-b413-c3ba7cf6de2f)

**Comment by aderusha** [comment]
I'm not sure if I follow the question now - what are you trying to do, and how isn't it working?

**Reply by mimk97** [reply]
i want to add a button to Page 3. 
so i split B4 and add B8 (see Screenshot)
then i deleted P11, cause Memory Full ( but it doesn´t matter, don´t need P11 :)
then i rotate Display 180 degrees and exported file as tft.
than flashed with webinterface
the Picture is ok. but i can´t touch it.
if i don´t rotate i can touch it

**Reply by aderusha** [reply]
OK I think I smell what you're cooking (I'll catch up eventually!) - I think what might be happening is that you're not sending events on the button.  Select b8, then make sure "Send Component ID" is checked for Touch Press Event and Touch Release Event:
![image](https://github.com/user-attachments/assets/cb6997cd-b5b0-4bda-b161-ff6b257ab350)

**Reply by mimk97** [reply]
send component ID is checked
the Touch Problem isn´t only B8
The Touch doesn´t work at all if i flash the inverted tft (my tft like i wrote before)
could you test it if i will send you the tft?

**Reply by aderusha** [reply]
What might be more helpful would be the debug logs - is the screen sending anything to the HASPone, and is the HASPone doing anything useful in response?

**Reply by mimk97** [reply]
how can i read this Debug Log?

**Reply by mimk97** [reply]
i thought it was this Problem with aligned Objects you wrote in this Thread?
https://github.com/HASwitchPlate/HASPone/discussions/155

**Reply by aderusha** [reply]
Debug logging: https://github.com/HASwitchPlate/HASPone/wiki/Capture-HASPone-debug-output

The thread you linked is more about the blueprints, some of which won't be working if you start modifying the base project.  How/when that happens is dependent on the specific blueprint, how it works, and the sorts of changes you've made.  This is something you will encounter, eventually, but probably isn't your issue right now.

**Reply by mimk97** [reply]
ah ok. Just for my understanding: 
I must only edit the HMI and Rotate the Display in Display Setting 180° and it must work?
not neccessary to edit something else in the Nextion Editor?
Cause to test i changed nothing in the HMI, just rotate and flash
Result is the same: Touch not working
If i flash the "unrotated" file, Touch works :(

**Reply by mimk97** [reply]
as an alternative Solution, do you know if it´s possible to get the Clock working to another Page. In my Case P9 B12

**Reply by aderusha** [reply]
Again, I don't really know what's happening but debug logging will tell you directly what the HASPone is seeing and how it's responding.  The clock blueprint prints text at a fixed size which won't work correctly with smaller buttons.  You can modify that blueprint to use a smaller font if that's the goal, but you might also run into limits on the size limit for the button you want to use, and also, the blueprint is doing some goofy stuff to provide an icon image next to the time which will have to be removed as it won't fit and also won't work.

**Reply by mimk97** [reply]
thanks. i think than i would do that. 
debug Log isn´t an Option for me. Casue the Screen is mounted in the Wall. Fixed with glue :)

**Reply by mimk97** [reply]
Thanks for your Help!

**Comment by aderusha** [comment]
> debug Log isn´t an Option for me. Casue the Screen is mounted in the Wall.

Check the linked wiki article above, telnet log is an option.  It's a really helpful look at what Nextion is doing and if you're going to be developing your own solutions, you're probably going to want to be able to look at what's happening.
